### PR TITLE
Nessie: Disable table metadata files cleanup during commits

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Avoid automatic cleanup of table metadata files during commit as it can be in use by other references.

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
cc: @nastra, @snazy, @dimas-b 

Since https://github.com/apache/iceberg/pull/7576 may take more time to conclude (or may not get merged). I have opened a catalog-specific PR targeted for the 1.3.0 release to avoid facing this problem for the users.

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java
Comment (dimas-b) [inline nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java:607]: Would it be more robust to parse `metadataFileLocation` as a URI, assert schema value, and call `.getPath()`?

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java
Comment (dimas-b) [inline nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java:598]: Is this expected to be overridden back to `false` by the Nessie Catalog? If so, shouldn't we assert that?

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java
Comment (ajantha-bhat) [reply nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java:613]: Can you paste the expected Assertions here? I think I didn't fully get it. 

Also, other places uses this metadata location string directly 
https://github.com/apache/iceberg/blob/39a5f503c1d19d41a15b74ea53e3d5e51d495874/hive-metastore/src/test/java/org/apache/iceberg/hive/HiveTableTest.java#L238

Comment (dimas-b) [reply nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java:613]: If it's a common pattern in test code, I'm fine with following it in Nessie tests.

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java
Comment (dimas-b) [inline nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java:621]: Would it be possible to do time travel in a Nessie-specific way to get the expected metadata location? Current test looks a bit magical. It asserts that some old metadata file exists, but it does not link it to Nessie behaviour.

Comment (ajantha-bhat) [reply nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java:622]: Nice suggestion. Added it.

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java
Comment (snazy) [inline nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java:97]: This log message is meaningless, nothing actionable, I find it confusing.

Comment (ajantha-bhat) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java:97]: If the user has set the table property and might expect it to work. So, checking logs can help in analyzing why it didn't work. 

Do you think printing a warning log only when the user has set this property is useful?

Comment (ajantha-bhat) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java]: Added conditional warning logs and improved logs to contain actionable item.

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java
Comment (ajantha-bhat) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java:104]: > I guess the WARN is going to be logged on every load, unless the user permanently resets the METADATA_DELETE_AFTER_COMMIT_ENABLED property to false, right?

No, this will be set in table metadata refresh flow. So, this property will be set to false and written to file in the first refresh.  So, it won't print on every load. For example, this testcase execution prints it only once.

Comment (ajantha-bhat) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java:104]: > How about Automatic table metadata files cleanup was requested, but disabled because the Nessie catalog can use historical metadata files. Please set the " + TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED + " property to 'false'. Use the 'nessie-gc' tool for history-aware GC".?

I don't think user has to reset it again (based on my above comment). So, I am thinking to update like this.

`Automatic table metadata files cleanup was requested, but disabled because the Nessie catalog can use historical metadata files from other references. Use the 'nessie-gc' tool for history-aware GC".`

Comment (dimas-b) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java:104]: SGTM :+1:

**Comment by ajantha-bhat** [comment]
@snazy, @dimas-b: Addressed the comments. Please take a look at it again.

**Review by Fokko — APPROVED** [review]

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java
Comment (ajantha-bhat) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java]: @snazy: We are waiting for your approval.

**Review by snazy — APPROVED** [review]

**Pull request merged**
